packages/headless/src/controllers/triggers/headless-redirection-trigger.ts

```diff
@@ -0,0 +1,106 @@
+import {Schema, Value} from '@coveo/bueno';
```

Comment:
Curious if there is a design doc for the trigger feature? Is the redirection trigger the only one in scope or is there a plan to support e.g. `notify` too?

Reply:
I am currently doing `redirection` but will work on `notify` and `query` once this is done.

Reply:
Ok, have the apis for each type been designed? Are they similar, or completely different? Depending on the answers, this will affect how things would best be named and structured.

If there is no design, I strongly recommend spending some time creating a document and sharing it with the team. It will us to collaborate and iterate faster compared to actually coding the feature,

Reply:
The `triggers.ts` file in headless defines the interfaces for each trigger.

Reply:
Ok, but will the controllers look like? Does it make sense to have a separate controller by type? Or would it be better to have one controller for everything related to triggers?

Reply:
I remember meeting with Olivier and Jean-Felix and deciding on having a controller for each trigger type but I don't remember the reason for this choice.